### PR TITLE
Add OfferGasReqBaseTest for testing gasreq and extend gasbase test for failure

### PR DESCRIPTION
**Pull request by lnist**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,9 +53,7 @@
     "solhint": "^3.6.2",
     "ts-node": "^10.9.1",
     "typescript": "^5.2.2",
-    "yargs": "^17.7.2",
-    "bn.js": "^5.2.1",
-    "mathjs": "^11.11.1"
```
Comment (espendk) [inline package.json:56]: Wasn't this just introduced  in the new bin tests?

File: test/core/gas/OfferGasBase.t.sol [hunk 2]
```diff
@@ -1,136 +1,7 @@
...
-  function test_gasbase_transfers_estimate() public {
-    uint outbound_gas = measureTransferGas($(base));
-    uint inbound_gas = measureTransferGas($(quote));
-    uint gasbase = 2 * (outbound_gas + inbound_gas);
-    console.log("Gas used: %s", gasbase);
-    description = string.concat(description, " - Case: ActivateSemibook transfers estimate");
-    printDescription();
-  }
-}
+import {OfferGasBaseBaseTest} from "mgv_test/lib/gas/OfferGasBaseBase.t.sol";
```
Comment (espendk) [inline test/core/gas/OfferGasBase.t.sol:4]: All your base are belong to us 😂

**Review by espendk — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (lnist) [reply package.json:56]: still there - just better ordering, I don't know why it was introduced out of order.

Comment (espendk) [reply package.json:56]: And why wasn't yarn.lock updated as a consequence?

File: test/core/gas/OfferGasBase.t.sol (hunk 2 above)
Comment (lnist) [reply test/core/gas/OfferGasBase.t.sol:4]: Yes! :)

**Pull request merged**